packages/esm-patient-test-results-app/src/withDashboardRouting.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from "react";
+import { Switch, Route, BrowserRouter, useParams } from "react-router-dom";
+
+export default function withDashboardRouting(WrappedComponent) {
+  const WrappedWithParams = (props) => {
+    const params = useParams();
+    return <WrappedComponent {...props} {...params} />;
+  };
+
+  return ({ basePath, patient, patientUuid }) => (
+    <BrowserRouter basename={`${window.spaBase}${basePath}/test-results`}>
```

Comment:
I think this whole component should change.

The `wrappedWithParams` is imho no longer required. Using the general navigation context also is now deprecated.

Reply:
okay, so what would be the proper alternative to extract parameter?

Reply:
The props are passed in. There is no need to extract parameters.

Reply:
Do you mean, that the complete is passed through as a prop and I can extract the parameters inside my component?
In need the uuids encoded in the URL in any case or the routing would not work.

Reply:
There is no URL anymore. You open the workspace by attaching an extension.

Reply:
This is not a workspace anymore but a dashboard (hence the name `withDashboardRouting`).
I used [esm-patient-notes-app](https://github.com/openmrs/openmrs-esm-patient-chart/blob/feature/monorepo/packages/esm-patient-notes-app/src/notes/notes.component.tsx) as reference.

Reply:
Then this was even the situation beforehand; props such as `patientUuid` as passed in directly. No need to do anything.

Reply:
The reason for the notes provider was that there are components within components that require complex objects such as the patient, which are not transported via the URL. While prop drilling could be done these more complex objects could also be transported via a context.